### PR TITLE
Drop 36lts code

### PR DESCRIPTION
**Pull request by clebergnu**
With 36lts now declared EOLed[1], let's remove code that was related to specifics of that release.

[1] - https://www.redhat.com/archives/avocado-devel/2018-February/msg00015.html

### CODE REVIEW

**Comment by clebergnu** [comment]
[Remove 36lts related code](https://trello.com/c/itptIpzB/1244-remove-36lts-related-code)

File: optional_plugins/varianter_yaml_to_mux/avocado_varianter_yaml_to_mux/mux.py [hunk 1]
```diff
@@ -181,9 +181,6 @@ def __iter__(self):
         if self.root is None:
             return
 
-        # TODO: Remove when 36lts is discontinued
-        if not hasattr(self, "variant_ids"):
-            self.variant_ids = self._get_variant_ids()
```
Comment (ldoktor) [inline optional_plugins/varianter_yaml_to_mux/avocado_varianter_yaml_to_mux/mux.py]: The function `_get_variant_ids` is used only once, perhaps it'd make sense to get rid of it completely and embed it into `initialize_mux`, what do you think?

File: avocado/core/varianter.py [hunk 2]
```diff
@@ -209,11 +209,6 @@ def _process_default_params(self, args):
             default_params.merge(default_param)
         self._default_params = default_params
         self.default_params.clear()     # We don't need these anymore
-        # FIXME: Backward compatibility params, to be removed when 36 LTS is
-        # discontinued
-        if (not getattr(args, "variants-skip-defaults", False) and
-                hasattr(args, "default_avocado_params")):
-            self._default_params.merge(args.default_avocado_params)
```
Comment (ldoktor) [inline avocado/core/varianter.py]: The same applies here, please consider embedding this code into `parse` as without the compatibility code it's just few lines related to parsing.

File: avocado/utils/build.py [hunk 3]
```diff
@@ -23,10 +23,11 @@ def run_make(path, make='make', env=None, extra_args='', ignore_status=None,
     """
     Run make, adding MAKEOPTS to the list of options.
 
+    :param path: directory from where to run make
     :param make: what make command name to use.
     :param env: dictionary with environment variables to be set before
                 calling make (e.g.: CFLAGS).
-    :param extra: extra command line arguments to pass to make.
+    :param extra_args: extra command line arguments to pass to make.
```
Comment (ldoktor) [inline avocado/utils/build.py]: Good, would you please also include the `ignore_status` description?

File: .travis.yml [hunk 4]
```diff
@@ -14,7 +14,6 @@ matrix:
 branches:
     only:
         - master
-        - 36lts
```
Comment (ldoktor) [inline .travis.yml]: this reminds me we should probably remove the `36lts` check from Avocado-vt's `.travis.yml` (it's used to check py2.6 branch).

File: avocado/core/parser.py [hunk 5]
```diff
@@ -135,9 +134,6 @@ def start(self):
         # Allow overriding default params by plugins
         variants = varianter.Varianter(getattr(self.args, "varianter_debug", False))
         self.args.avocado_variants = variants
-        # FIXME: Backward compatibility params, to be removed when 36 LTS is
-        # discontinued
-        self.args.default_avocado_params = tree.TreeNode()
```
Comment (ldoktor) [inline avocado/core/parser.py]: Before doing this we have to fix `avocado-virt`, which is still using the `default_avocado_params`.

File: optional_plugins/varianter_yaml_to_mux/avocado_varianter_yaml_to_mux/mux.py (hunk 1 above)
Comment (clebergnu) [reply optional_plugins/varianter_yaml_to_mux/avocado_varianter_yaml_to_mux/mux.py]: I don't have a problem with that.  Adding that change in a new commit.

File: avocado/utils/build.py (hunk 3 above)
Comment (clebergnu) [reply avocado/utils/build.py]: Sure, adding.

File: .travis.yml (hunk 4 above)
Comment (clebergnu) [reply .travis.yml]: True!

File: avocado/core/parser.py (hunk 5 above)
Comment (clebergnu) [reply avocado/core/parser.py]: Before merging this, sure.  I'm heading to review your fixes for Avocado-Virt.

**Comment by clebergnu** [comment]
Closing in favor of #2460

**Pull request closed without merging**